publish_courses/api.py

```diff
@@ -159,171 +147,109 @@ def create(self, request, *args, **kwargs) -> Response:
-                    'DataType': 'String'
-                },
-            }
-
-            # Append the message to the list of messages for this batch.
-            messages.append({
-                'Id': message_id,
-                'MessageBody': message_body,
-                'MessageAttributes': message_attributes
-            })
-
-        return messages
-
-    def send_sqs_message_batch(self, messages: List[Dict], sqs_msg_batch_size: int = 10) -> None:
+        logger.info(f'Details for the bulk publish courses job saved to database', extra={
+                    'job_id': job_id})
+        
+        return None 
+        
+    def send_course_list_to_lambda(self, job_id: str, selected_courses: List) -> None:
         """
-        Sends a batch of messages to an Amazon Simple Queue Service (SQS) queue.
-
-        Note: As of 2023-11-02 you can use send_message_batch to send up to 10 messages to the specified queue.
-        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs/client/send_message_batch.html
+        Invokes SQS queueing Lambda with a payload of course ID list.
+        This is a fire and forget method and will not wait for the Lambda to finish.
         """
-        logger.debug(
-            f'Bulk publish courses sending messages to SQS queue name: "{QUEUE_NAME}"')
+        logger.info(f'Sending bulk publish courses job {job_id} to queuing Lambda {QUEUEING_LAMBDA_NAME}')
         
         try:
-            queue = SQS.get_queue_url(QueueName=QUEUE_NAME)['QueueUrl']
-
-            for i in range(0, len(messages), sqs_msg_batch_size):
-                batch = messages[i:i + sqs_msg_batch_size]
-                response = SQS.send_message_batch(
-                    QueueUrl=queue,
-                    Entries=batch
-                )
-
-                context = {
-                    'batch': batch,
-                    'response': response
-                }
-                logger.debug(
-                    f'Bulk publish courses job sent to SQS', extra=context)
+            # Create Lambda client
+            lambda_client = boto3.client('lambda')
+        except ClientError as e:
+            logger.error(f'Error configuring Lambda client: {e}', exc_info=True)
+            raise
         except Exception as e:
-            logger.exception(f"Error sending SQS message batch: {e}")
+            logger.exception('Error configuring Lambda client')
             raise
+
+        # Construct the payload.
+        payload = json.dumps({
+            "job_id": job_id,
+            "queue_name": QUEUE_NAME,
+            "course_id_list": selected_courses
+        })
+
+        # Invoke Lambda function.
+        response = lambda_client.invoke(
+            FunctionName=QUEUEING_LAMBDA_NAME,
+            InvocationType='Event',
+            Payload=payload
+        )
+
+        logger.info(f'Sent bulk publish courses job {job_id} to queuing Lambda {QUEUEING_LAMBDA_NAME}',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You have a log statement that might use unsanitized input. Depending on the context, this could result in:

1. A log injection attack that breaks log integrity, forges log entries, or bypasses monitors that use the logs. To increase the security of your code, sanitize your inputs before logging them. [Learn more](https://cwe.mitre.org/data/definitions/117.html)

2. A sensitive information leak that exposes users&#39; credentials, private information, or identifying information to an attacker. To preserve privacy in your code, redact sensitive user information before logging it. [Learn more](https://cwe.mitre.org/data/definitions/532.html)